src/test/java/edu/harvard/iq/dataverse/externaltools/ExternalToolHandlerTest.java

```diff
@@ -198,4 +205,33 @@ public void testGetToolUrlWithOptionalQueryParameters() {
 
     }
 
+    @Test
+    public void testGetToolUrlWithallowedApiCalls() {
+
+        System.out.println("allowedApiCalls test");
+        Dataset ds = new Dataset();
+        ds.setId(1L);
+        ApiToken at = new ApiToken();
+        AuthenticatedUser au = new AuthenticatedUser();
+        au.setUserIdentifier("dataverseAdmin");
+        at.setAuthenticatedUser(au);
+        at.setTokenString("1234");
+        ExternalTool et = ExternalToolServiceBeanTest.getAllowedApiCallsTool();
+        assertTrue(et != null);
+        System.out.println("allowedApiCalls et created");
+        System.out.println(et.getAllowedApiCalls());
+        ExternalToolHandler externalToolHandler = new ExternalToolHandler(et, ds, at, null);
+        System.out.println("allowedApiCalls eth created");
+        JsonObject jo = externalToolHandler
+                .createPostBody(externalToolHandler.getParams(JsonUtil.getJsonObject(et.getToolParameters()))).build();
+        assertEquals(1, jo.getJsonObject("queryParameters").getInt("datasetId"));
+        String signedUrl = jo.getJsonArray("signedUrls").getJsonObject(0).getString("signedUrl");
+        // The date and token will change each time but check for the constant parts of
+        // the response
+        assertTrue(signedUrl.contains("https://librascholar.org/api/v1/datasets/1"));
```

Comment:
You need to add          System.setProperty(SITE_URL, "https://librascholar.org"); to get this test to pass

Reply:
May I drop a hint about https://guides.dataverse.org/en/latest/developers/testing.html#junit-5-test-helper-extensions here? It's very easy to transform this test to a JUnit 5 testcase to make use of this little helpers that take care about handling the global state...

Reply:
Fixed. FWIW, this test worked when all are run because one of the exporter tests sets this. (It could/should use @SystemProperty as well.)